Source/EasyNetQ/MessagePropertiesExtensions.cs

```diff
@@ -0,0 +1,80 @@
+using RabbitMQ.Client;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace EasyNetQ;
+
+/// <summary>
+///     Represents various properties of a message
+/// </summary>
+internal static class MessagePropertiesExtensions
+{
+    public const string ConfirmationIdHeader = "EasyNetQ.Confirmation.Id";
+
+    public static MessageProperties SetConfirmationId(this MessageProperties properties, ulong confirmationId)
+    {
+        properties.Headers[ConfirmationIdHeader] = confirmationId.ToString();
+        return properties;
+    }
+
+    public static bool TryGetConfirmationId(this MessageProperties properties, out ulong confirmationId)
+    {
+        confirmationId = 0;
+        return properties.Headers.TryGetValue(ConfirmationIdHeader, out var value) &&
+               ulong.TryParse(Encoding.UTF8.GetString(value as byte[] ?? Array.Empty<byte>()), out confirmationId);
+    }
+
+    public static void CopyFrom(this MessageProperties source, IBasicProperties basicProperties)
```

Comment:
`public static void CopyFrom(this MessageProperties source, IBasicProperties basicProperties)`
->
`  public static MessageProperties CopyFrom(this MessageProperties source, IBasicProperties basicProperties)`

Reply:
Even that
 ```
public static MessageProperties ToMessageProperties(this IBasicProperties basicProperties)
{  
    var messageProperties = new MessageProperties();
    messageProperties.CopyFrom(basicProperties);
    return messageProperties;
}
```
since in most cases you call `CopyFrom` on newly created message properties.